### PR TITLE
Split the agent into separate containers

### PR DESCRIPTION
**Pull request by craigbr**
Created new containers agent-collector and agent-forwarder. They
use the agent-base container and just layer on specific files for
running the collector and forwarder respectively. Log files go to
stdout.

Remove the old monasca-agent Dockerfile, etc. The new containers
should be used instead.

Update agent-base README for AGENT_USER arg

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -4,8 +4,15 @@ services:
   influxdb:
     image: influxdb:latest
 
-  agent:
-    image: monasca/agent:latest
+  agent-forwarder:
+    image: craigbryant/agent-forwarder:latest
```
Comment (mhoppal) [inline docker-compose.yml]: should this be monasca?

File: monasca-agent-base/README.md [hunk 2]
```diff
@@ -55,6 +55,7 @@ A few build argument can be set:
    be a tagged point release (e.g. `1.6.0`), an OpenStack release branch (e.g.
    `stable/newton`), a Gerrit patch ref (e.g. `refs/changes/71/427271/10`),
    or any other valid Git ref for the target repository.
+ * `AGENT_USER`: the user used to run the forwarder
```
Comment (mhoppal) [inline monasca-agent-base/README.md]: why would this matter in a container?

File: monasca-agent-collector/README.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
-monasca-agent Dockerfile
+monasca-agent-collector Dockerfile
 ========================
 
-This image contains a containerized version of the Monasca agent. For more
-information on the Monasca project, see [the wiki][1].
+This image contains a containerized version of the Monasca agent collector. For
+more information on the Monasca project, see [the wiki][1]. It is based on the
```
Comment (mhoppal) [inline monasca-agent-collector/README.md]: should you point to the monasca agent wiki?

File: monasca-agent-collector/README.md [hunk 4]
```diff
@@ -27,15 +28,15 @@ released or merged. Until this changes, only `master` images may be available.
 Usage
 -----
 
-The agent requires at least a reachable [Monasca API][5] server. Access to
-additional services, like the Kubernetes API, is also necessary if they are to
+The agent collector requires at least a reachable [monasca-collector][5]. Access
```
Comment (mhoppal) [inline monasca-agent-collector/README.md]: forward?

File: monasca-agent-collector/README.md [hunk 5]
```diff
@@ -195,32 +197,32 @@ processed as Jinja2 templates with access to all environment variables.
 Building
 --------
 
-To build the container from scratch, run:
+[dbuild][10] can be used with the build.yml file to build and push the
+container.
+
+To build the container from scratch using just docker commands, run:
 
     docker build -t youruser/monasca-agent:latest .
```
Comment (mhoppal) [inline monasca-agent-collector/README.md]: collector?

File: monasca-agent-forwarder/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,13 @@
+FROM monasca/agent-base:master-20170614-225502
+
+ARG AGENT_USER="mon-agent"
```
Comment (mhoppal) [inline monasca-agent-forwarder/Dockerfile]: again does this need to be configurable

File: monasca-agent-forwarder/README.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+Usage
+-----
+
+The agent forwarder requires a reachable [Monasca API][5] server.
+
+In environments resembling the official [docker-compose][3] or [Kubernetes][6]
+environments, the image does not require additional configuration parameters and
+can be minimally run like so:
+
+    docker run -it monasca/agent:latest
```
Comment (mhoppal) [inline monasca-agent-forwarder/README.md]: monasca-agent-forwarder?

File: monasca-agent-forwarder/README.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+
+ * Docker container monitoring: On a plain Docker host, collects standard
+   system-level metrics about each running container (e.g. cpu, memory, etc).
+   Requires `DOCKER=true`.
+ * Kubernetes container monitoring: Collects container metrics for a particular
+   node from cAdvisor. Intended to be run on each node as a `DaemonSet`.
+   Requires `KUBERNETES=true`.
+ * Kubernetes API monitoring: Collects metrics about a Kubernetes cluster,
+   including health status, node disk/memory pressure, capacities, etc.
+   Requires `KUBERNETES_API=true`.
```
Comment (mhoppal) [inline monasca-agent-forwarder/README.md]: wouldnt all this be on the collector not forwarder?

File: monasca-agent-collector/README.md [hunk 9]
```diff
@@ -195,32 +197,32 @@ processed as Jinja2 templates with access to all environment variables.
...
 [3]: https://github.com/hpcloud-mon/monasca-docker/
 [4]: https://github.com/hpcloud-mon/monasca-docker/blob/master/monasca-agent/Dockerfile
-[5]: https://hub.docker.com/r/monasca/api/
+[5]: https://hub.docker.com/r/monasca/agent-forwarder/
 [6]: https://github.com/hpcloud-mon/monasca-docker/blob/master/k8s/
 [7]: https://kubernetes.io/docs/user-guide/downward-api/
 [8]: https://github.com/hpcloud-mon/monasca-docker/blob/master/monasca-agent/agent.yaml.j2
 [9]: https://github.com/google/cadvisor#quick-start-running-cadvisor-in-a-docker-container
 [10]: https://github.com/timothyb89/git-sync
+[11]: https://github.com/timothyb89/git-sync
```
Comment (mhoppal) [inline monasca-agent-collector/README.md]: same link twice?

Comment (timothyb89) [inline monasca-agent-collector/README.md]: Looks like `[10]` is pointing to the wrong place, actually.

**Pull request merged**